connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/TestPlugins.java

```diff
@@ -64,56 +63,95 @@
+        // on the plugin path is undeterministic. So, hide the v2 jar from the plugin path. This jar
+        // is only used in certain testcases that verify resources are read correctly when provided
+        // in both the PluginClassLoader and its parent class loader.
+        READ_VERSION_FROM_RESOURCE_V2("read-version-from-resource-v2",
+                "test.plugins.ReadVersionFromResource", true),
 
-    /**
-     * Class name of a plugin which will always throw an exception during loading
-     */
-    public static final String ALWAYS_THROW_EXCEPTION = "test.plugins.AlwaysThrowException";
-    /**
-     * Class name of a plugin which samples information about its initialization.
-     */
-    public static final String ALIASED_STATIC_FIELD = "test.plugins.AliasedStaticField";
-    /**
-     * Class name of a {@link org.apache.kafka.connect.storage.Converter}
-     * which samples information about its method calls.
-     */
-    public static final String SAMPLING_CONVERTER = "test.plugins.SamplingConverter";
-    /**
-     * Class name of a {@link org.apache.kafka.common.Configurable}
-     * which samples information about its method calls.
-     */
-    public static final String SAMPLING_CONFIGURABLE = "test.plugins.SamplingConfigurable";
-    /**
-     * Class name of a {@link org.apache.kafka.connect.storage.HeaderConverter}
-     * which samples information about its method calls.
-     */
-    public static final String SAMPLING_HEADER_CONVERTER = "test.plugins.SamplingHeaderConverter";
-    /**
-     * Class name of a {@link org.apache.kafka.common.config.provider.ConfigProvider}
-     * which samples information about its method calls.
-     */
-    public static final String SAMPLING_CONFIG_PROVIDER = "test.plugins.SamplingConfigProvider";
-    /**
-     * Class name of a plugin which uses a {@link java.util.ServiceLoader}
-     * to load internal classes, and samples information about their initialization.
-     */
-    public static final String SERVICE_LOADER = "test.plugins.ServiceLoaderPlugin";
+        /**
+         * Resource dir and class names of plugins that are included in a single jar.
+         */
+        MULTIPLE_PLUGINS_IN_JAR_THING_ONE("multiple-plugins-in-jar", "test.plugins.ThingOne"),
+        MULTIPLE_PLUGINS_IN_JAR_THING_TWO("multiple-plugins-in-jar", "test.plugins.ThingTwo");
+
+        private final String resourceDir;
+        private final String className;
+        private final boolean hideFromPluginPath;
+
+        TestPlugin(String resourceDir, String className) {
+            this(resourceDir, className, false);
+        }
+
+        TestPlugin(String resourceDir, String className, boolean hideFromPluginPath) {
+            this.resourceDir = resourceDir;
+            this.className = className;
+            this.hideFromPluginPath = hideFromPluginPath;
+        }
+
+        public String getResourceDir() {
```

Comment:
Nit: we don't use the `get` prefix in this code base:

```suggestion
        public String resourceDir() {
```

Also, this method is unused at the moment.